sklearn/pipeline.py

```diff
@@ -1102,20 +1102,20 @@ def __sklearn_is_fitted__(self):
             return False
 
     def _sk_visual_block_(self):
-        _, estimators = zip(*self.steps)
-
         def _get_name(name, est):
             if est is None or est == "passthrough":
                 return f"{name}: passthrough"
             # Is an estimator
             return f"{name}: {est.__class__.__name__}"
 
-        names = [_get_name(name, est) for name, est in self.steps]
+        names, estimators = zip(
+            *[(_get_name(name, est), est) for name, est in self.steps]
+        )
         name_details = [str(est) for est in estimators]
         return _VisualBlock(
             "serial",
             estimators,
-            names=names,
+            names=list(names),
```

Comment:
Having to add this partially undid the idea of removing one loop. :shrug: While I still think the refactor is an _ever-so-slight_ improvement, I'm OK if want to close it, too.